File: ios/Anemone SDK/model/tenant/TenantType.swift
//
//  TenantType.swift
//  Anemone SDK
//
//  Created by Manuel González Villegas on 26/6/18.
//  Copyright © 2018 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public enum TenantType: String {
    case syltekcrm
    case playtomicIntegrated = "playtomic_integrated"
    case anemoneOnly = "anemone"

    static func fromRawValue(_ rawValue: String) -> TenantType? {
        TenantType(rawValue: rawValue.lowercased())
    }
}

extension TenantType: CustomStringConvertible {
    public var description: String {
        rawValue
    }
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/model/tenant/TenantType.kt
package com.anemonesdk.model.tenant

import com.playtomic.foundation.model.CustomStringConvertible

enum class TenantType constructor(var rawValue: String) : CustomStringConvertible {
    syltekcrm("syltekcrm"),
    playtomicIntegrated("playtomic_integrated"),
    anemone("anemone");

    companion object {

        fun fromRawValue(rawValue: String): TenantType? = entries.firstOrNull { it.rawValue == rawValue.lowercase() }
    }

    override val description get() = rawValue
}
